lib/tasks/sonarqube_scan.rake

```diff
@@ -0,0 +1,24 @@
+require 'rake'
+
+namespace :sonar do
+  desc "Run a sonarqube scan"
+  task :scan => :environment do
+    raise "Test coverage required, Run 'CI=1 rake spec' to generate coverage/.resultset.json" unless File.exist?("coverage/.resultset.json")
+    raise "Scanner jar required, download from here https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-4.2.0.1873.zip then set the env var with SCANNER_PATH=/path/to/sonar-scanner/lib/sonar-scanner-cli-4.2.0.1873.jar" unless ENV['SCANNER_PATH']
+
+    setup unless File.exist?("sonar-scanner.properties")
+
+    system("java -jar #{ENV['SCANNER_PATH']} -Dproject.settings=sonar-scanner.properties")
+  end
+
+  def setup
+    raise "TOKEN needed to upload to sonar server" unless ENV['TOKEN']
+
+    user = ENV['USER'] || `whoami`
+    token = ENV['TOKEN']
```

Comment:
Should we be more specific with the env variable names?  Maybe: `SONAR_PATH` and `SONAR_TOKEN`.